composer/utils/object_store/uc_object_store.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+import logging
+import os
+import pathlib
+import uuid
+from typing import Callable
+
+from composer.utils.import_helpers import MissingConditionalImportError
+from composer.utils.object_store.object_store import ObjectStore
+
+log = logging.getLogger(__name__)
+
+
+class UCObjectStore(ObjectStore):
+    """Utility class for uploading and downloading data from Databricks Unity Catalog Volumes
+    """
+
+    def __init__(self, uri: str) -> None:
+        try:
+            import databricks
+        except ImportError as e:
+            raise MissingConditionalImportError('databricks') from e
+
+        self.prefix = self._get_prefix(uri)
+
+        if not 'DATABRICKS_HOST' in os.environ or 'DATABRICKS_TOKEN' not in os.environ:
+            # TODO: Raise a better exception here
+            raise ValueError('Environment variables `DATABRICKS_HOST` and `DATABRICKS_TOKEN` '
+                             'must be set to use Databricks Unity Catalog Volumes')
+
+        from databricks.sdk import WorkspaceClient
+        self.client = WorkspaceClient()
+
+    @staticmethod
+    def _get_prefix(uri: str) -> str:
+        # removeprefix works only with python3.9 and above
+        if hasattr(uri, 'removeprefix'):
+            return uri.removeprefix('uc:/')
+        else:
+            if uri.startswith('uc:/'):
+                return uri[len('uc:/'):]
+            return uri
+
+    def get_uri(self, object_name: str) -> str:
+        return f'uc:/{self.get_object_path(object_name)}'
```

Comment:
This is because `get_object_path(..)` returns something like `/Volumes/...` so `get_uri(..)` will return `uc://Volumes/...`

Reply:
After the update, this will be of the format `dbfs:/Volumes/...` which is the expected format for dbfs files. I believe it is `dbfs:/` instead of `dbfs://` because it used to follow hadoop file format conventions although the latest hadoop url formats do except ``://` from what I can tell

Reply:
ok i think this works...
```
In [3]: parse_uri('dbfs:/Volumes/...')
Out[3]: ('dbfs', '', 'Volumes/...')
```
And I guess this makes sense because otherwise we think there is a bucket
```
In [4]: parse_uri('dbfs://Volumes/...')
Out[4]: ('dbfs', 'Volumes', '...')
```